vllm/core/block/prefix_caching_block.py

```diff
@@ -65,10 +69,10 @@ def __init__(
         # We share the refcounter between allocators. This allows us to promote
         # blocks originally allocated in the hashless allocator to immutable
         # blocks.
-        self._refcounter = self._hashless_allocator.refcounter
+        self.refcounter = self._hashless_allocator.refcounter
```

Comment:
Can we avoid making `refcounter` part of the public API? We can access `.refcounter` on the `_hashless_allocator` because this implementation knows that _hashless_allocator is of type NaiveBlockAllocator. We don't need to expose `refcounter` for all BlockAllocators -- it expands the API without any benefit.

Let me know if I miss something.

Reply:
This is not surfaced to `BlockAllocator`. It just says `BlockAllocator` and `PrefixCachingBlockAllocator` has `refcounter`, which helps the type checker.

Reply:
why not just keep the underscore then?

Reply:
fixed in [edac6b4](https://github.com/vllm-project/vllm/pull/5968/commits/edac6b473c43a99ad382e577827be7ad17061bf5)